apidef/graphql_config_adapter.go

```diff
@@ -0,0 +1,280 @@
+package apidef
+
+import (
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"github.com/jensneuse/graphql-go-tools/pkg/ast"
+	"github.com/jensneuse/graphql-go-tools/pkg/astparser"
+	graphqlDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/graphql_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/httpclient"
+	restDataSource "github.com/jensneuse/graphql-go-tools/pkg/engine/datasource/rest_datasource"
+	"github.com/jensneuse/graphql-go-tools/pkg/engine/plan"
+	"github.com/jensneuse/graphql-go-tools/pkg/graphql"
+)
+
+var ErrUnsupportedGraphQLConfigVersion = errors.New("provided version of GraphQL config is not supported for this operation")
+
+type GraphQLConfigAdapter struct {
+	config     GraphQLConfig
+	httpClient *http.Client
+}
+
+func NewGraphQLConfigAdapter(config GraphQLConfig) GraphQLConfigAdapter {
```

Comment:
What do u think about rename adapter to smth like `GraphQLConfigV2Adapter` as this whole adapter is responsible for creating v2 engine config

in this way, we could omit v2 prefixes for method names

Reply:
The adapter adapts a single input (GraphQLConfig) into different outputs (currently only v2, but could be more in the future)

Reply:
Maybe, in this case, it will be better to have an adapters package and different implementations for the different kinds of configs?

Reply:
moved into own package